src/components/QueryBuilder.vue

```diff
@@ -49,7 +50,32 @@ export default Vue.extend( {
 		};
 	},
 	methods: {
+		validate(): boolean {
+			this.errors = [];
+			this.$store.dispatch( 'setErrors', [] );
```

Comment:
```ts
this.errors = [];
this.$store.dispatch( 'setErrors', [] );
```
Those two lines together are strange. Since we are mapping the state to `errors`, shouldn't be one enough?

Reply:
IIRC, this is an issue with proxies in Vue2 (setting arrays) and can be removed in Vue3.

Reply:
We should look into migrating to Vue3. Probably, better sooner than later.

Reply:
I think it was a deliberate choice to go with vue2, I think it was because of consistency across the whole wmde codebase (and we can't migrate mediawiki micron frontends yet). Jakob knows the details.